src/coreneuron/CMakeLists.txt

```diff
@@ -444,6 +440,13 @@ endif()
 # https://forums.developer.nvidia.com/t/cannot-dynamically-load-a-shared-library-containing-both-openacc-and-cuda-code/210972
 add_library(coreneuron-core STATIC ${CORENEURON_CODE_FILES} ${CORENRN_MPI_OBJ})
 add_dependencies(coreneuron-core coreneuron-copy-nrnivmodl-core-dependencies)
+if(CORENRN_ENABLE_OPENMP)
+  target_link_libraries(coreneuron-core PUBLIC OpenMP::OpenMP_CXX)
+  # The above uses a generator expression to arrive at -fopenmp (or equivalent) - this will not work
+  # with our Makefile generation. Thus specify the flag manually. Remove once we get rid of the
+  # mechanism building Makefile.
+  target_compile_options(coreneuron-core PUBLIC ${OpenMP_CXX_FLAGS})
```

Comment:
Thanks for the useful comment

Reply:
Does this mean we get `g++ -fopenmp -fopenmp`? Can the issue be solved closer to where the problem occurs? The issue seems to be that when we extract flags, when we encounter `OpenMP::OpenMP_CXX` we do the "wrong" thing. Maybe that code can be made to handle the OpenMP target correctly?

Reply:
Yes.

Thinking about it, I can for now comment out the target usage (updating the comment). That would also allow to simplify the code for the flag extraction.

Reply:
Regarding using `OpenMP::OpenMP_CXX` and doing the right thing: it uses a generator expression in the compile options, which does not work with our Makefile generation. That is going to be very hard to hack around in the right way.

I've now disabled the target linking and instead use the libraries and compile options as above. This works with my CMake export branch, too. Once we drop the Makefile, we can go target-only.

Reply:
This is strange, I can see you removed the extra complexity from the flag extraction code, but AFAICT, it's still linking against the `OPENMP::OPENMP_CXX` target.

Reply:
nvm, github is showing me outdated code.